src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -2408,41 +2405,30 @@ private Matches() {}
   public static Match<UnitType> unitTypeCanBeInBattle(final boolean attack, final boolean isLandBattle,
       final PlayerID player, final int battleRound, final boolean includeAttackersThatCanNotMove,
       final boolean doNotIncludeAa, final boolean doNotIncludeBombardingSeaUnits) {
-    return Match.of(ut -> {
-      // Filter out anything like factories, or units that have no combat ability AND cannot be taken casualty
-      final Match.CompositeBuilder<UnitType> canBeInBattle = Match.newCompositeBuilder();
-
-      final Match.CompositeBuilder<UnitType> supportOrNotInfrastructureOrAa = Match.newCompositeBuilder();
-      supportOrNotInfrastructureOrAa.add(unitTypeIsInfrastructure().invert());
-      supportOrNotInfrastructureOrAa.add(unitTypeIsSupporterOrHasCombatAbility(attack, player));
-      if (!doNotIncludeAa) {
-        supportOrNotInfrastructureOrAa.add(Match.allOf(unitTypeIsAaForCombatOnly(),
-            unitTypeIsAaThatCanFireOnRound(battleRound)));
-      }
-      canBeInBattle.add(supportOrNotInfrastructureOrAa.any());
-
-      if (attack) {
-        if (!includeAttackersThatCanNotMove) {
-          canBeInBattle.add(unitTypeCanNotMoveDuringCombatMove().invert());
-          canBeInBattle.add(unitTypeCanMove(player));
-        }
-        if (isLandBattle) {
-          if (doNotIncludeBombardingSeaUnits) {
-            canBeInBattle.add(unitTypeIsSea().invert());
-          }
-        } else { // is sea battle
-          canBeInBattle.add(unitTypeIsLand().invert());
-        }
-      } else { // defense
-        if (isLandBattle) {
-          canBeInBattle.add(unitTypeIsSea().invert());
-        } else { // is sea battle
-          canBeInBattle.add(unitTypeIsLand().invert());
+
+    // Filter out anything like factories, or units that have no combat ability AND cannot be taken casualty
+    final PredicateBuilder<UnitType> canBeInBattle = PredicateBuilder.of(unitTypeIsInfrastructure().invert())
```

Comment:
Nit: Might want to rename this `canBeInBattleBuilder`.  I had to take a double-take below when I saw statements like:

```java
canBeInBattle
    .and(unitTypeCanNotMoveDuringCombatMove().invert())
    .and(unitTypeCanMove(player));
```

and I didn't see the result being assigned back to `canBeInBattle` (i.e. I confused it for a `Predicate` instead of a `PredicateBuilder`).